examples/erc20/src/lib.rs

```diff
@@ -0,0 +1,267 @@
+        }
+        Err(Error::InsufficientFunds { address: from })
+    }
+
+    /// allowance
+    pub fn approve(&mut self, ctx: &Context, spender: Address, amount: u64) -> Result<Approval> {
+        let allowances = match self.allowed.entry(ctx.sender()) {
+            Entry::Vacant(ve) => ve.insert(HashMap::new()),
+            Entry::Occupied(oe) => oe.into_mut(),
+        };
+        allowances.insert(spender, amount);
+
+        let approval = Approval {
+            sender: ctx.sender(),
+            spender,
+            amount,
+        };
+
+        Event::emit(&approval);
+
+        Ok(approval)
+    }
+
+    /// read allowance
+    pub fn allowance(&mut self, ctx: &Context, spender: Address) -> Result<u64> {
+        if !self.allowed.contains_key(&ctx.sender()) {
+            return Ok(0u64);
+        }
+        Ok(self
+            .allowed
+            .get(&ctx.sender())
+            .and_then(|allowances| allowances.get(&spender))
+            .copied()
+            .unwrap_or_default())
+    }
+
+    /// transfer from a given account up to the given allowance
+    pub fn transfer_from(
+        &mut self,
+        _ctx: &Context,
+        from: Address,
+        spender: Address,
+        amount: u64,
+    ) -> Result<Transfer> {
+        let allowances = self.allowed.get_mut(&from).unwrap();
+        // if the spender is not in the list of addresses that are approved for automatic
+        // withdrawal by the from address, then nothing can be done
+        if !allowances.contains_key(&spender) {
+            return Err(Error::NoAllowanceGiven { from, to: spender });
+        }
+        let allowance = allowances.get(&spender).copied().unwrap_or_default();
+        // err if request is higher than allowance
+        if allowance < amount {
+            return Err(Error::RequestExceedsAllowance { amount, allowance });
+        }
+        if do_transfer(&mut self.accounts, from, spender, amount) {
+            allowances.insert(spender, allowance - amount);
+            return Ok(Transfer {
+                from,
+                to: spender,
+                amount,
+            });
+        }
+        Err(Error::InsufficientFunds { address: from })
+    }
+}
+
+impl ERC20Token {
+    /// mint new tokens
+    pub fn mint(&mut self, ctx: &Context, amount: u64) -> Result<()> {
+        if !self.admins.contains(&ctx.sender()) {
+            return Err(Error::AdminPrivilegesRequired);
+        }
+        self.total_supply = self.total_supply + amount;
+        Ok(())
+    }
+
+    /// burn tokens from a given account
+    pub fn burn(&mut self, ctx: &Context, from: Address, amount: u64) -> Result<()> {
+        if !self.admins.contains(&ctx.sender()) {
+            return Err(Error::AdminPrivilegesRequired);
+        }
+        let balance = self.accounts.get(&from).copied().unwrap_or_default();
+        self.accounts
+            .insert(from, std::cmp::max(0, balance - amount));
+        Ok(())
+    }
+}
+
```

Comment:
where's your
```rust
fn main() {
	mantle::service!(ERC20Token);
}
```
?

Reply:
Done. Moved `lib.rs` to `main.rs` as well.